lib/src/common/disposable.dart

```diff
@@ -26,19 +26,50 @@ abstract class _Disposable {
   Future<Null> dispose();
 }
 
-class InternalDisposable implements _Disposable {
+/// Used to invoke, and remove references to, a [Disposer] before disposal of the
+/// parent object.
+class ManagedDisposer implements _Disposable {
```

Comment:
I think it sill provides value. It provides a common interface between `ManagedDisposer` and `Disposable`, this allows us to store both items in (and type) the `_internalDisposables` list in `Disposer`. We could make `Disposable` implement `ManagedDisposer`, I had that earlier today and it led to some confusion  exactly  the object was and how to document it.